src/replica/replication_app_base.cpp

```diff
@@ -45,107 +45,26 @@ namespace replication {
 
 error_code replica_init_info::load(const std::string &dir)
 {
-    std::string old_info_path = utils::filesystem::path_combine(dir, ".info");
-    std::string new_info_path = utils::filesystem::path_combine(dir, ".init-info");
-    std::string load_path;
-    error_code err;
-    if (utils::filesystem::path_exists(new_info_path)) {
-        load_path = new_info_path;
-        err = load_json(new_info_path.c_str());
-    } else {
-        load_path = old_info_path;
-        err = load_binary(old_info_path.c_str());
-    }
-
-    ERR_LOG_AND_RETURN_NOT_OK(err, "load replica_init_info from {} failed", load_path);
-
-    ddebug_f("load replica_init_info from {} succeed: {}", load_path, to_string());
-    if (load_path == old_info_path) {
-        // change replica_init_info file from old to new
-        ERR_LOG_AND_RETURN_NOT_OK(store_json(new_info_path.c_str()),
-                                  "change replica_init_info file from {} to {} failed",
-                                  old_info_path,
-                                  new_info_path);
-
-        ddebug_f(
-            "change replica_init_info file from {} to {} succeed", old_info_path, new_info_path);
-        dassert_f(
-            utils::filesystem::remove_path(old_info_path), "remove file {} failed", old_info_path);
-    }
-
+    std::string info_path = utils::filesystem::path_combine(dir, ".init-info");
```

Comment:
I suggest `.init-info` defined as a const variable, which will more safe when someone to update it, like https://github.com/XiaoMi/rdsn/blob/6af2c3204c889a83dd0767383ec60ea65f089a9b/src/replica/replica.cpp#L53

Reply:
+1